knotx-adapter/knotx-adapter-common/src/main/java/io/knotx/adapter/common/http/HttpClientFacade.java

```diff
@@ -49,19 +50,40 @@
 
   private final WebClient webClient;
 
-  public HttpClientFacade(WebClient webClient, List<ServiceMetadata> services) {
+  private final JsonObject customRequestHeader;
+
+  public HttpClientFacade(WebClient webClient, HttpAdapterConfiguration configuration) {
     this.webClient = webClient;
-    this.services = services;
+    this.services = configuration.getServices();
+    this.customRequestHeader = configuration.getCustomRequestHeader();
   }
 
   public Single<ClientResponse> process(AdapterRequest message, HttpMethod method) {
     return Single.just(message)
         .doOnSuccess(this::validateContract)
         .map(this::prepareRequestData)
-        .flatMap(serviceRequest -> callService(serviceRequest, method))
+        .flatMap(
+            serviceRequest -> callService(serviceRequest, method)
+                .doOnSuccess(resp -> logResponse(serviceRequest, resp)))
         .flatMap(this::wrapResponse);
   }
 
+  private void logResponse(Pair<ClientRequest, ServiceMetadata> request,
+                           HttpResponse<Buffer> resp) {
+    if (resp.statusCode() >= 400 && resp.statusCode() < 600) {
```

Comment:
Do we want to `trace` other status codes?

Reply:
Good point. I'll check and will add trace for it and then do the merge